File: C++/Parser.cpp
#include "Parser.h"

/**
 * Constructor. Opens file to be parsed
 */
Parser::Parser(std::string f)
{
    file.open(f.c_str(), std::ifstream::in);
    if (!file.is_open()) {
        exit(1);
    }
}

/**
 * Destructor. Destroys vector and closes file
 */
Parser::~Parser()
{
    teams.empty();
    file.close();
}

/**
 * Called to create the vector of teams
 */
std::vector<Team> Parser::create_teams()
{
    std::string input;
    while (!file.eof()) {
        Team team = create_team();
        add_team(team);
    }
    return teams;
}

/**
 * Private function. Creates one team and returns it
 */
Team Parser::create_team()
{
    std::string name = "";
    int _asm, _to, _dto, _pendif, _rettd;
    double _pya, _dpya, _rya, _drya;
    file >> name >> _asm >> _pya >> _dpya >> _rya >> _drya >> _to >> _dto >> _pendif >> _rettd;
    if (!name.empty())
        return Team(name, _asm, _pya, _dpya, _rya, _drya, _to, _dto, _pendif, _rettd);
}

/**
 * add_team(Team) receives a new team to add to teams. First checks if the new_team is unique,
 * then adds to the teams vector
 */
void Parser::add_team(Team& new_team)
{
    bool team_is_unique = true;
    for (int i = 0; i < teams.size(); i++) {
        if (teams[i].getName().compare(new_team.getName()) == 0) {
            team_is_unique = false;
            break;
        }
    }
    if (team_is_unique)
        teams.push_back(new_team);
}

File: Java/edu/tamu/csce315/Parser.java
package edu.tamu.csce315;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

/**
 * Created by Cory Thompson on 1/20/16. <br>
 * This class is designated to read the input file and parse the information to create the Teams.
 */
public class Parser {
    private Scanner scanner;

    /**
     * Constructor. Opens file and prepares for Parsing
     * @param file String name of file to open
     */
    public Parser(String file) {
        try {
            scanner = new Scanner(new File(file));
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
    }

    /**
     * Creates the array of teams by reading in from the file
     * @return ArrayList of Teams
     */
    public ArrayList<Team> createTeams() {
        ArrayList<Team> teams = new ArrayList<>();
        while (scanner.hasNext()) {
            teams.add(createTeam());
        }
        return teams;
    }

    /**
     * Private function to create one team by reading in one line of the file
     * @return Team
     */
    private Team createTeam() {
        String name = scanner.next();
        int asm     = scanner.nextInt();
        double pya  = scanner.nextDouble();
        double dpya = scanner.nextDouble();
        double rya  = scanner.nextDouble();
        double drya = scanner.nextDouble();
        int to      = scanner.nextInt();
        int dto     = scanner.nextInt();
        int pendif  = scanner.nextInt();
        int rettd   = scanner.nextInt();
        return new Team(name, asm, pya, dpya, rya, drya, to, dto, pendif, rettd);
    }
}
